scripts/build.sh

```diff
@@ -0,0 +1,293 @@
+    else
+        echo 'PUSH_WINDOWS not set, not including Windows in manifest'
+        sed -e "s|TAG|$1|g" \
+        -e "s|REGISTRY|$REGISTRY|g" \
+        -e '/^WIN_ONLY/d' \
+        manifest_spec.yaml.tmpl > ./build/manifest_spec.yaml;
+    fi
+}
+
+push_manifest_with_tag() {
+    gen_manifest_with_tag $1
+	./manifest-tool push from-spec ./build/manifest_spec.yaml
+}
+
+push() {
+	# Assumes you have the images built or loaded already. Not
+	# added as dependency due to having both Linux/Windows
+	# prereqs which can't be done on the same machine.
+    if [ $PUSH_WINDOWS ] ; then
+        for arch in $WIN_ARCH; do
+            push_images TARGET="sonobuoy-win-$arch"
+        done
+    else
+        echo 'PUSH_WINDOWS not set, not pushing Windows images'
+        for arch in $LINUX_ARCH; do
+            push_images TARGET="sonobuoy-arch"
+        done
+    fi
+
+	push_manifest_with_tag $IMAGE_BRANCH
+	push_manifest_with_tag $GIT_REF_SHORT
+
+	if [ git describe --tags --exact-match >/dev/null 2>&1 ] ; then
+		push_manifest_with_tag $IMAGE_VERSION
+		push_manifest_with_tag latest
+		push_manifest_with_tag $IMAGE_TAG
+	fi
+}
+
+remove-image() {
+	docker rmi -f `docker images $REGISTRY/$1 -a -q` || true
+}
+
+clean() {
+    # Best effort for clean; don't exit if failure.
+    set +e
+	rm -f $TARGET
+	rm Dockerfile-*
+	rm DockerfileWindows-*
+	rm -rf build
+
+	for arch in $LINUX_ARCH; do
+		remove-image $TARGET-$arch
+	done
+	for arch in $WIN_ARCH; do
+		remove-image $TARGET-win-$arch
+	done
+    remove-image $TARGET
+    set -e
+}
+
+# kind_images will build the kind-node image. Generally building the base image is not necessary
+# and we can use the upstream kindest/base image.
+kind_images() {
+    K8S_PATH=$GOPATH/src/github.com/kubernetes/kubernetes
+    KIND_K8S_TAG=$(cd $K8S_PATH && git describe)
+	kind build node-image --kube-root=$K8S_PATH --image $REGISTRY/kind-node:$KIND_K8S_TAG
+}
+
+# push_kind_images will push the same image kind_images just built our registry.
+push_kind_images() {
+    K8S_PATH=$GOPATH/src/github.com/kubernetes/kubernetes
+    KIND_K8S_TAG=$(cd $K8S_PATH && git describe)
+	docker push $REGISTRY/kind-node:$KIND_K8S_TAG
+}
+
+# check-kind-env will show you what will be built/tagged before doing so with kind_images
+check-kind-env() {
+    if [ -z $K8S_PATH ] ; then
+        $error K8S_PATH is undefined
+    fi
+    if [ -z $KIND_K8S_TAG ] ; then
+        echo KIND_K8S_TAG is undefined
+    fi
+	echo --kube-root=$K8S_PATH tagging as --image $REGISTRY/kind-node:$KIND_K8S_TAG
+}
+
+setup_kind_cluster(){
+    if ! kind get clusters | grep -q "^$KIND_CLUSTER$"; then
+        kind create cluster --name "$KIND_CLUSTER" --config kind-config.yaml
+        # Although the cluster has been created, not all the pods in kube-system are created/available
+        sleep 20
+    fi
+}
+
+# High level function; composes functions above and additional functionality. Typically just used by CI.
+prepare_and_run_integration() {
```

Comment:
This function takes over what the run_integration_tests.sh used to do. Seems like it belongs here now. The only reason it was split off was because make was a pain I think :)